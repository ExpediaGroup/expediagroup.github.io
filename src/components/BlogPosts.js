/*
Copyright 2022 Expedia, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import styles from './BlogPosts.module.css';

function BlogPosts({posts}) {
    return (
        <div className={styles.posts}>
            { posts.map(post => (
                <BlogPost key={post.title} post={post}/>
            )) }
        </div>
    );
}

function BlogPost({post}) {
    return (
        <div className={styles.post}>
            <div className={styles.postHeader}>
                <a href={post.link} target="_blank">
                    <h4>{post.title}</h4>
                </a>
            </div>
            <div className={styles.postBody}>
                <span className={styles.creator}>{post.creator}</span>
                <span className={styles.date}>{post.date}</span>
            </div>
        </div>
    );
}

export default BlogPosts;
