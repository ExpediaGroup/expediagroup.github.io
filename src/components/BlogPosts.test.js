/*
Copyright 2022 Expedia, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import renderer from 'react-test-renderer';
import BlogPosts from './BlogPosts';

const POSTS = [{
    title: 'My Post Title',
    creator: 'Mary Krismass',
    link: 'https://my/post',
    date: 'Jan 01, 2022',
    imageUrl: 'https://my/post/image.jpg'
}];

it('renders correctly', () => {
    const tree = renderer
        .create(<BlogPosts posts={POSTS}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
