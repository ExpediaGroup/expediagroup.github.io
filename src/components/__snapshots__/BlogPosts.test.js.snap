// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="posts"
>
  <div
    className="post"
  >
    <div
      className="postImageColumn"
    >
      <a
        href="https://my/post"
        target="_blank"
      >
        <img
          src="https://my/post/image.jpg"
        />
      </a>
    </div>
    <div
      className="postDetailColumn"
    >
      <div
        className="postTitle"
      >
        <a
          href="https://my/post"
          target="_blank"
        >
          <h4>
            My Post Title
          </h4>
        </a>
      </div>
      <div
        className="postCreationInfo"
      >
        <span
          className="creator"
        >
          Mary Krismass
        </span>
        <span
          className="date"
        >
          Jan 01, 2022
        </span>
      </div>
    </div>
  </div>
</div>
`;
